### ISSUE TITLE
MethodNotFound exceptions when using with latest HtmlAgilityPack

### ISSUE THREAD

**Issue by brutaldev** [atifaziz/Fizzler#56]
Fizzler does not work with the latest version of HtmlAgilityPack (1.4.9.5) even though the NuGet package allows it. It only works with 1.4.9 and no versions ahead of that.  Developers of HtmlAgilityPack have broken compatibility with "revision" builds so they can be blamed for this...

**Comment by atifaziz** [comment]
Thanks for bringing this to my attention.

> Developers of HtmlAgilityPack have broken compatibility with "revision" builds so they can be blamed for this…

That's really too bad. Do you have the stack trace that tells which method has been removed?

**Comment by brutaldev** [comment]
```
'System.MissingMethodException' in System.Core.dll
Method not found: 'System.Collections.Generic.IEnumerable`1<HtmlAgilityPack.HtmlNode> HtmlAgilityPack.HtmlNode.Descendants()
```
I just reverted the version when I saw this so never took a close look. The error in System.Core means it might actually be them, except that it only happens when you call `QuerySelectorAll`

**Comment by jwbats** [comment]
If the fault was with Fizzler, I'd expect the author to have fixed it by now.

[I've just notified Html Agility Pack of this.](https://htmlagilitypack.codeplex.com/workitem/43511)

Both Fizzler and HAP are now aware.

**Comment by atifaziz** [comment]
@jwbats Thanks for reporting the issue to HAP. You've got my vote on it. :) I would add, however, that the issue title is not representative of the problem and may not warrant anyone's attention. Clearly, the problem is that there was a breaking change in HAP (removal of one or more methods) without increasing the major version number. This probably can be reproduced without Fizzler. One can then mention in passing that this is affecting users of projects like Fizzler that use HAP.

Also, do you absolutely need to be on the “breaking” version of HAP? If not, I would suggest simply revert to the last working version & pin your dependency. I've noticed some [renewed activity in the project on CodePlex](http://htmlagilitypack.codeplex.com/SourceControl/list/changesets), including several new versions published to NuGet ([1.4.9.1](https://www.nuget.org/packages/HtmlAgilityPack/1.4.9.1)&hellip;[1.4.9.5](https://www.nuget.org/packages/HtmlAgilityPack/1.4.9.5) so far this year) but can't understand the real use cases for the changes and which HAP issues were the drivers.

**Comment by jwbats** [comment]
I have edited my issue on codeplex to more accurately represent the actual problem.

I do not *need* to be on the latest version and am indeed using 1.4.9 without problems currently. But I'm such a big fan of HAP + Fizzler, I would hate to see this issue lying around for longer than it already has.

**Comment by JohnBergman** [comment]
FWIW, there are other packages that do not use Fizzler, but use HAP that have the issue.  Dialogue, an Umbraco plugin is one such example

**Comment by brutaldev** [comment]
This should be working now, they added the old parameterless method back specifically for Fizzler.

https://stackoverflow.com/a/44913952/564726
https://github.com/zzzprojects/html-agility-pack/issues/4
https://github.com/zzzprojects/html-agility-pack/blob/3db9cd80976a2c888240b51ae2c8872fea7bcddc/src/HtmlAgilityPack/HtmlNode.cs#L973

**Comment by atifaziz** [comment]
@brutaldev Thanks for dropping a note on that fix. Hopefully HAP 1.5 is out soon and I can then go ahead and update the dependency to line up with that version.

**Comment by brutaldev** [comment]
@atifaziz They are already on 1.5.3. This was fixed in 1.5.0-beta5 so it only affected those few dot releases between 1.4.9 and the final of 1.5.0. I saw they were actively developing again so I was going to ask if they could fix this and after some digging it looks like they already did.

You might want to change your NuGet package to ignore the breaking 1.4.9.x dependency or just update it to > 1.5.0
